src/table.rs

```diff
@@ -152,11 +163,28 @@ impl Table {
     ///
     /// To insert to table, use `entry` to return a mutable reference
     /// and set it to the appropriate value.
-    pub fn entry<'a>(&'a mut self, key: &str) -> &'a mut Item {
-        let parsed_key = key.parse::<Key>().expect("invalid key");
-        &mut self
+    pub fn entry<'a>(&'a mut self, key_str: &str) -> &'a mut Item {
+        let parsed_key = key_str.parse::<Key>().expect("invalid key");
+        let path = parsed_key.get_key_path();
+        let key = &path[path.len() - 1];
+
+        let table = if parsed_key.is_dotted_key() {
+            // Insert the marker for dotted key.
+            self.items
+                .entry(parsed_key.raw().to_owned())
+                .or_insert(TableKeyValue::new(
+                    key_repr(parsed_key.raw()),
+                    Item::DottedKeyMarker(parsed_key.parts.clone())
+                ));
+            TomlParser::descend_path(self, &path[..path.len() - 1], 0, true)
+                .expect("the table path is valid; qed")
```

Comment:
hmm, looking at this function it always return `Ok` (or panics).
Looks like something was lost in a refactoring.